### PR TITLE
chore(integrations): auth mode consts for connStatus funcs

### PR DESCRIPTION
**Pull request by daabr**

### CODE REVIEW

**Review by RonenMars — APPROVED** [review]
LGTM and I'll be glad you expose the `integrations/auth.go` file using gRPC so that we can match and validate our values with the frontend, that way, we will protect ourselves (not 100%) from having different **authTypes** in use on frontend and the backend.

File: integrations/auth.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package integrations
+
+// Align these with:
+// https://github.com/autokitteh/web-platform/blob/main/src/enums/connections/connectionTypes.enum.ts
+const (
+	APIToken   = "apiToken"
+	JSONKey    = "jsonKey" // Google integrations only
+	OAuth      = "oauth"
+	PAT        = "pat"
+	SocketMode = "socketMode" // Slack integration only
```
Comment (pashafateev) [inline integrations/auth.go:10]: I noticed that not all types found [here](https://github.com/autokitteh/web-platform/blob/main/src/enums/connections/connectionTypes.enum.ts) are represented. Is this intentional?

**Review by pashafateev — APPROVED** [review]

**Comment by daabr** [comment]
> LGTM and I'll be glad you expose the `integrations/auth.go` file using gRPC so that we can match and validate our values with the frontend, that way, we will protect ourselves (not 100%) from having different **authTypes** in use on frontend and the backend.

Let's discuss offline, I don't think a gRPC endpoint is valuable.

File: integrations/auth.go (hunk 1 above)
Comment (daabr) [reply integrations/auth.go:10]: Yes, the list in this PR is (or rather will be) the authoritative one. The TS file is tentative.

**Pull request merged**